core/templates/components/profile-link-directives/profile-link-image-backend-api.service.ts

```diff
@@ -33,8 +33,10 @@ export class ProfileLinkImageBackendApiService {
   constructor(
     private http: HttpClient
   ) {}
-
-  async fetchProfilePictureDataAsync(profileImageUrl: string): Promise<string> {
+  // It will return a 'null' Promise when the user is not logged in or has not
+  // uploaded a profile picture, or the player is in preview mode.
```

Comment:
[optional]
```suggestion
  // It will return a 'null' Promise in the following cases:
  // 1. The user is not logged in
  // 2. The user didn't uploaded a profile picture
  // 3. The player is in preview mode.
```

Reply:
I think this is more neat, thanks @mridul-netizen!